### PR TITLE
added productListing page to website

### LINKED ISSUE #11: adding productListing page
adding products to productpage

### PR DESCRIPTION
**Pull request by mariyasada**
added products to productListing page
create a separate component for product card
productListing

### CODE REVIEW

File: src/context/product-context.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { useContext, useState,createContext,useEffect} from "react";
+
+const ProductContext =createContext();
+ const ProductProvider =({children})=>{
+    const [productList,setProductList]=useState([]);
+
+    useEffect(()=>{
+        (async()=>{
+          try{
+              const {data:{products}} = await axios.get("/api/products")
```
Comment (Raghav888) [inline src/context/product-context.js:11]: Nice, you have used destructuring here

**Review by Raghav888 — APPROVED** [review]

File: src/components/ProductCard/Productcard.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { Link } from 'react-router-dom';
+import { FaStar,FaRegHeart,FaRupeeSign } from "react-icons/fa";
+import "../ProductCard/Productcard.css";
+
+
+export const Productcard = ({userProduct}) => {
+  return (
```
Comment (JantuDeb) [inline src/components/ProductCard/Productcard.jsx:9]: I would suggest you to destructure userProduct here.
```
const {isSoldOut, brand, productName} = userProduct
```

File: src/components/ProductList/ProductList.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Productcard } from '../ProductCard/Productcard';
+import "../ProductList/ProductList.css";
+
+ export const ProductList = () => {
+     const {productList,setProductList}=useProduct();
+  return (
+    <div className="product-content-container">
+      <div className="grid-3-column-layout">
+          {productList.map((product)=>{
+              return <Productcard userProduct={product}/>
```
Comment (JantuDeb) [inline src/components/ProductList/ProductList.jsx:12]: I think you can name this prop  `products`
```
<ProductCard products={products} >

```
This is just a suggestion.

**Review by JantuDeb — APPROVED** [review]
Toh coded this really well I have added comments in some places other than that this is really good.

File: src/components/ProductCard/Productcard.jsx (hunk 2 above)
Comment (mariyasada) [reply src/components/ProductCard/Productcard.jsx:9]: done

File: src/components/ProductList/ProductList.jsx (hunk 3 above)
Comment (mariyasada) [reply src/components/ProductList/ProductList.jsx:12]: will do this also

**Pull request merged**